cql3/statements/alter_session_prop_defs.cc

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright (C) 2020 ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "alter_session_prop_defs.hh"
+#include "types.hh"
+#include "concrete_types.hh"
+
+void cql3::statements::alter_session_prop_defs::validate() {
+    static const std::set<sstring> keywords({"latency_limit_for_reads", "latency_limit_for_writes"});
```

Comment:
We might (or not) want to recommend that if you have different latency goals for reads and writes, you should use different sessions.

Reply:
(if we do, then there's be just latency, not latency for read and latency for write).

Reply:
Yup, may make sense. I split it up in this RFC just because reads and writes do usually have totally different latencies in metrics, but it's all arbitrary for now. Users probably have a single latency requirement and don't care if the underlying operation is a read, a write or something in-between.

Reply:
Actually, there was one more reason I split the values - scylla config has separate timeouts for these two. Is it Cassandra legacy which we can ignore, or do users perhaps sometimes set up different configs for these two?